File: Forget-Api/Forget.Core.Application/Services/CategoryService.cs
using AutoMapper;
using Forget.Core.Application.Core;
using Forget.Core.Domain;
using Forget.Core.Service.Contracts;
using Forget.Core.Service.Core;
using Forget.Core.Service.Dtos.Category;
using Forget.Core.Service.Repositories;

namespace Forget.Core.Application.Services;

public class CategoryService: GenericService<CategoryDto, SaveCategoryDto, Category>, ICategoryService
{
  private readonly ICategoryRepository _repository;
  private readonly IMapper _mapper;
  public CategoryService(ICategoryRepository repository, IMapper mapper) : base(repository, mapper)
  {
    _repository = repository;
    _mapper = mapper;
  }
}

File: Forget-Api-Ts/src/services/category.service.ts
import { injectable } from 'tsyringe'
import { GenericService } from '../core'
import { ICategoryService } from '../interfaces/services'
import { Category, CategoryModel } from '../models'

@injectable()
export class CategoryService extends GenericService<Category> implements ICategoryService {
  constructor () {
    super(CategoryModel)
  }
}
